### ISSUE TITLE
Execution/install issues

### ISSUE THREAD

**Issue by cryptovestor21** [anyblockanalytics/thegraph-allocation-optimization#1]
Hey guys, first of all kudos for developing this in the open, I hope other like-minded Indexers also contribute to this valuable tool.

I had a go at installing this on a ubuntu 20.04.2 LTS vm. I had issues installing python-glpk, the package doesn't seem to exist.

When I run the script against the example indexer_id I get the current state, but optimisation fails (looks like because the above package is missing)
```
$ python3 ./allocation_script.py --indexer_id 0x453b5e165cf98ff60167ccd3560ebf8d436ca86c --max_percentage 0.9 --threshold 20 --parallel_allocations 4
Script Execution on:  29042021_18:05:42

(stuff about current allocations here)..... 

Optimize Allocations:
==============================

Optimize Allocations for Interval: indexingRewardDay and Percentage per Allocation: 0.9
WARNING: Could not locate the 'glpsol' executable, which is required for
    solver 'glpk'
Traceback (most recent call last):
  File "./allocation_script.py", line 541, in <module>
    solver.solve(model)
  File "/home/node/.local/lib/python3.8/site-packages/pyomo/opt/base/solvers.py", line 518, in solve
    self.available(exception_flag=True)
  File "/home/node/.local/lib/python3.8/site-packages/pyomo/opt/solver/shellcmd.py", line 118, in available
    raise ApplicationError(msg % self.name)
pyutilib.common._exceptions.ApplicationError: No executable found for solver 'glpk'
```

When I run the script against **my own** indexer_id I get differing errors:
```
$ python3 ./allocation_script.py --indexer_id 0xC430be492ddEb6e761dbBd0E08baFe99F5064d90 --max_percentage 0.9 --threshold 20 --parallel_allocations 1
Script Execution on:  29042021_17:57:44
Traceback (most recent call last):
  File "./allocation_script.py", line 386, in <module>
    indexer_total_allocations = int(indexer_data.get('allocatedTokens')) * 10 ** -18
AttributeError: 'NoneType' object has no attribute 'get'
```

My address is a contract, same as the one in the example so not sure what the problem is there. Would be grateful for any pointers on where this might be going wrong.

**Comment by cryptovestor21** [comment]
OK I fixed the issue with my own address - addresses need to be all lowercase.

**Comment by enderYm** [comment]
Hey, sorry for the late response. Could you fix all your issues? Or is there still something that doesn't work?

**Comment by cryptovestor21** [comment]
I haven't found a fix for the missing package issue.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by enderYm** [comment]
Ok. Does `glpk -h` return an error? And did `sudo apt-get install glpk-utils` work without problems? Also are you using any kind of virtual environment for python?

**Comment by cryptovestor21** [comment]
> Ok. Does `glpk -h` return an error? And did `sudo apt-get install glpk-utils` work without problems? Also are you using any kind of virtual environment for python?

Installing glpk-utils fixed it! Thank you. I am not sure why this did not work the first few times I tried, the repo was reporting that the package didn't exist at that point but it appears to be there now.

Now I can run the tool against your address and it works, and I noticed that in order to have it work for my own address, I had to convert the address to lower case or it would throw an error.
